[Tag] Adds support for url prop (#4837)
Shopify/polaris: src/components/Tag/README.md

<Tag onClick={() => console.log('Clicked')}>Wholesale</Tag>
```

> ### Tagss with links

Reviewer: ```suggestion
### Tag with link
```

> Use to allow merchants to navigate to an entity.

Reviewer: ```suggestion
Use to allow merchants to navigate to a resource, like a customer segment or a smart collection.
```

Reviewer 2: ```suggestion
Use to allow merchants to navigate to a resource. For example a customer segment or a smart collection
```

Building on Chloe's suggestion a little

        mollis consectetur eros, quis maximus lorem dignissim at. Proin in
        rhoncus massa. Vivamus lectus nunc, fringilla euismod risus commodo,
        mattis blandit nulla
      </Tag>
      <Tag url="https://www.shopify.com/" onRemove={() => {}}>
        Wholesale
      </Tag>
      <Tag url="https://www.shopify.com/" onRemove={() => {}}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer at
        ipsum quam. Aliquam fermentum bibendum vestibulum. Vestibulum
        condimentum luctus metus, sed sagittis magna pellentesque eget. Duis
        dapibus pretium nisi, et venenatis tortor dignissim ut. Quisque eget
        lacus ac ex eleifend ultrices. Phasellus facilisis ex sit amet leo
        elementum condimentum. Ut vel maximus felis. Etiam eget diam eu eros
        blandit interdum. Sed eu metus sed justo aliquam iaculis ac sit amet ex.
        Curabitur justo magna, porttitor non pulvinar eu, malesuada at leo. Cras
        mollis consectetur eros, quis maximus lorem dignissim at. Proin in
        rhoncus massa. Vivamus lectus nunc, fringilla euismod risus commodo,
        mattis blandit nulla
      </Tag>

>     </Stack>

Reviewer: ```suggestion
    <Tag url="/collections/wholesale">Wholesale</Tag>;
```

  );
}
```

### Tag with custom content

> Use to allow merchants to generate special custom content for a tag.

Reviewer: ```suggestion
Use when a tag needs to be visually distinguished from others, like when it's added automatically.
```

      <Icon source={SearchMinor} />
      <span>Order #123</span>
      <Badge status="attention" progress="partiallyComplete">
        In Progress
      </Badge>
    </Stack>
  </Tag>

  <Tag onClick={() => console.log('Clicked')} title="Order #123">
    <Stack alignment="center" spacing="extraTight">
      <Icon source={SearchMinor} />
      <span>Order #123</span>
      <Badge status="attention" progress="partiallyComplete">
        In Progress
      </Badge>
    </Stack>
  </Tag>

  <Tag onRemove={() => console.log('Removed')} title="Order #123">
    <Stack alignment="center" spacing="extraTight">
      <Icon source={SearchMinor} />
      <span>Order #123</span>
      <Badge status="attention" progress="partiallyComplete">
        In Progress
      </Badge>
    </Stack>
  </Tag>

  <Tag
    url="https://www.shopify.com/"
    onRemove={() => console.log('Removed')}
    title="Order #123"
  >
    <Stack alignment="center" spacing="extraTight">
      <Icon source={SearchMinor} />
      <span>Order #123</span>
      <Badge status="attention" progress="partiallyComplete">
        In Progress
      </Badge>
    </Stack>
  </Tag>

> </Stack>

Reviewer: We try to keep the examples simple and reference real use cases when possible, so when `WandMinor` ships in `@shopify/polaris-icons` we should update this to reflect the real admin use case. We def want to test worst case/edge case scenarios, but we can do this in the playground 💯  

```suggestion
<Tag url="/customers?tag=VIP">
  <Stack spacing="extraTight">
    <Icon source={CustomersMinor} />
    <span>VIP</span>
  </Stack>
</Tag>
```
